crates/storage/src/sync.rs

```diff
@@ -0,0 +1,84 @@
+//! Synchronisation utilities for external runtimes.
+
+use core::cell::RefCell;
+use std::io;
+
+use borsh::{to_vec, BorshDeserialize, BorshSerialize};
+
+use crate::env;
+use crate::integration::Comparison;
+use crate::interface::Action;
+
+/// An artifact to aid synchronisation with an external runtime.
+#[derive(Debug, BorshSerialize)]
+pub enum SyncArtifact {
+    /// A list of actions.
+    Actions(Vec<Action>),
+    /// A list of comparisons.
+    Comparisons(Vec<Comparison>),
+}
+
+impl BorshDeserialize for SyncArtifact {
+    fn deserialize_reader<R: io::Read>(reader: &mut R) -> io::Result<Self> {
+        let Ok(tag) = u8::deserialize_reader(reader) else {
+            return Ok(SyncArtifact::Comparisons(vec![]));
+        };
+
+        match tag {
+            0 => Ok(SyncArtifact::Actions(Vec::deserialize_reader(reader)?)),
```

Comment:
make magic numbers enums?

Reply:
it technically "is" a modified form of what `BorshDeserialize` does for enums, but we have to do it manually here since sync initialization won't include any information, and borsh has no way to automatically account for that.

Reply:
I mean maybe just wrap the type_id into a CRDTTypeID enum